sdk/src/keystore/mod.rs

```diff
@@ -44,14 +44,133 @@ pub mod error;
 /// Schnorrkel Support
 pub mod sr25519;
 
-use alloc::string::ToString;
+use crate::clients::tangle::runtime::TangleConfig;
+use crate::keystore::sp_core_subxt::crypto::{DeriveError, SecretStringError};
+use crate::keystore::sp_core_subxt::DeriveJunction;
+#[cfg(any(feature = "std", feature = "wasm"))]
+// TODO: Once subxt uses sp-core 34.0.0, we can simply use sp_core
+pub use crate::tangle_subxt::subxt::ext::sp_core as sp_core_subxt;
+use alloy_signer_local::LocalSigner;
 use eigensdk_rs::eigen_utils::crypto::bls::{self as bls_bn254, g1_point_to_g1_projective};
 pub use error::Error;
-use schnorrkel::SignatureError;
-use subxt_core::config::PolkadotConfig;
-use subxt_core::tx::signer::PairSigner;
+use k256::ecdsa::SigningKey;
+use subxt::ext::sp_core::Pair as PairSubxt;
+use subxt_core::tx::signer::{PairSigner, Signer};
+use subxt_core::utils::{AccountId32, MultiAddress, MultiSignature};
+#[cfg(any(feature = "std", feature = "wasm"))]
+use tangle_subxt::subxt;
 
-pub type TanglePairSigner = PairSigner<PolkadotConfig, subxt_core::ext::sp_core::sr25519::Pair>;
+#[cfg(any(feature = "std", feature = "wasm"))]
+#[derive(Clone, Debug)]
+pub struct TanglePairSigner<Pair> {
```

Comment:
Why do we need this at all? I'm confused, it's a wrapper over something that works and we're reimplementing all the traits just to call and prefix it with `Tangle`?

Reply:
It's a wrapper designed to be used throughout the codebase to minimize common issues that can arise when internal/external developers work with keys. To ensure consistency, it limits freedom by enforcing structure, which we lacked before.